Ui tweaks

somexpert opened the pull request:
This branch deals mostly with the user interface, such as adding pagination to the home page, adding separation to the event details tables, monospacing the event ID's, etc.
Fixes #13 
Fixes #21 
Fixes #22 
Fixes #23 
Fixes #24 
Fixes #25

somexpert commented:
Here is the how the new home page looks.
![image](https://cloud.githubusercontent.com/assets/5580991/8209736/499747d0-14d4-11e5-9898-411d4f0c1595.png)

somexpert commented:
And this is the updated event details page.
![image](https://cloud.githubusercontent.com/assets/5580991/8209755/60db1926-14d4-11e5-8db1-157fdd1f98ea.png)

somexpert commented:
Think it's ready for review @ldko and @damonkelley

damonkelley commented on major_event_log/templates/major-event-log/index.html:
```diff
@@ -6,28 +6,71 @@
+            </tbody>
+        </table>
+        {% if is_paginated %}
+            <div class="pagination">
```
Maybe try adding `.pagination` to the `<ul>` element below. That we allow you to remove the custom `.pagination` definition in `major-event-log.css` in favor of Bootstrap's definition.

somexpert replied:
Didn't realize that existed. I'll try it out.

damonkelley replied:
Yup. Here is a link to pagination component docs.
http://getbootstrap.com/components/#pagination

damonkelley commented on major_event_log/templates/major-event-log/index.html:
```diff
@@ -6,28 +6,71 @@
+            {% endfor %}
+            </tbody>
+        </table>
+        {% if is_paginated %}
```
I prefer to put the pagination in a partial that I include in other templates. While you are currently only using pagination in a single template and there is no need to for re-usability at the present, pulling out that putting the pagination _component_ in it's own file makes it a bit easier to inspect templates later on IMO.

Just a suggestion.

somexpert replied:
Ah, yes, I noticed others did that too. I wasn't sure it was justified in this case because, as you mentioned, it's only used in one template, but I see your reasoning for preferring it separated. That's easy enough to do.

Commit:
Removed truncation from event IDs on the home page, as well as removing the time from the event date on the home page.

damonkelley commented:
Also, did you happen to see the Code subsection of the Typography docs in the Bootstrap documentation? http://getbootstrap.com/css/#code

The styling applied to the `<samp>` does what your `.monospaced` class does. It also has fallback font-faces.

somexpert commented:
Well I do remember looking at that, but it didn't click that it was exactly what I needed. Just ignored it since it was directed at showing code. But it makes sense.

damonkelley commented:
### With `.monospaced`

![selection_025](https://cloud.githubusercontent.com/assets/4121849/8210711/de4a7d62-14d8-11e5-91db-f9e0effb85a4.png)

### With `<samp>`

![selection_026](https://cloud.githubusercontent.com/assets/4121849/8210736/f9bbdc44-14d8-11e5-91c8-0063d110b5c8.png)

##### `<samp>` Fallback fonts

![selection_027](https://cloud.githubusercontent.com/assets/4121849/8210748/0abbd134-14d9-11e5-98ba-acc9b5894c59.png)

I believe there is the _slightest_ difference, probably because I have Monaco or Consolas available on my system.

damonkelley commented:
Also, that would mean you can remove your static files :wink:

somexpert commented:
.....seems like I've done that before..... lol. But yeah, I'll make that change and remove the static files.

damonkelley commented on major_event_log/templates/major-event-log/index.html:
```diff
@@ -6,28 +6,71 @@
+                    <td class="monospaced"><span class="label label-success">{{ event.get_outcome_display }}</span></td>
+                    {% else %}
+                    <td class="monospaced"><span class="label label-danger">{{ event.get_outcome_display }}</span></td>
+                    {% endif %}
```
We can strip this if block down to a single element with

``` html
<td>
    <span class="label {% if event.get_outcome_display == 'Success' %}label-success{% else %}label-danger{% endif %}>...</span>
</td>
```

And if you really wanted to keep it DRY you could do

``` html
<td>
    <span class="label label-{% if event.get_outcome_display == 'Success' %}success{% else %}danger{% endif %}>...</span>
</td>
```

Probably want to open up a new issue to add a `is_success` method to the Event model so that we don't need to  do this

``` html
{% if even.get_outcome_display == 'Success' %}
```

somexpert replied:
Yeah I remember you mentioning that to me last week. I'll add that to the issues.

Commit:
Used bootstraps pagination styling, separated pagination into its own file, used <samp> to remove monospaced class, removed static files, and created and used the is_succes() method within the model. Fixes #32, fixes #31, and the previous commit fixed #30 and fixed #29.

Commit:
Needed to do this merge because I accidentally pushed a single commit from a
copy of my local repo, rather than the repo I've been working on all along. Merge branch 'ui-tweaks' of https://github.com/unt-libraries/django-major-event-log into ui-tweaks

somexpert commented:
Home page now looks like this:
![image](https://cloud.githubusercontent.com/assets/5580991/8214642/33e122f0-14f0-11e5-8e3f-8e36a4cb8efb.png)

somexpert commented:
Event details page now looks like this:
![image](https://cloud.githubusercontent.com/assets/5580991/8214652/4563ead0-14f0-11e5-9f65-a71a3622d37e.png)

vphill commented:
The updated UI look and feel wise looks good to me.  I don't have direct comments on the coding of the UI though.

ldko commented:
:+1:

damonkelley commented on major_event_log/models.py:
```diff
@@ -30,6 +30,12 @@ class Event(models.Model):
     contact_email = models.EmailField()
 
+    def is_success(self):
+        if self.get_outcome_display() == "Success":
```
It would be wise for us to rework the `OUTCOME_CHOICES` for this

Currently we have 

``` python
OUTCOME_CHOICES = (
    ("http://purl.org/NET/UNTL/vocabularies/eventOutcomes/#success",
     "Success"),
    ("http://purl.org/NET/UNTL/vocabularies/eventOutcomes/#failure",
     "Failure"),
)
```

Which means that `"Success"` is not contained in a single location. Furthermore, I am not sure how I feel about comparing the labels. Comparing the value of `outcome` seems to be more appropriate.

Transforming the code above into this will allow us to do this easily.

``` python
SUCCESS = "http://purl.org/NET/UNTL/vocabularies/eventOutcomes/#success"
FAILURE = "http://purl.org/NET/UNTL/vocabularies/eventOutcomes/#failure"
OUTCOME_CHOICES = (
    (SUCESS,  "Success"),
    (FAILURE, "Failure"),
)
```

Then you can compare `SUCCESS` to the value of `outcome` to determine if it is successful or not.

Does that make sense?

somexpert replied:
I like it. I did try comparing the value of outcome to the uri, but that seemed a bit messy. This will clean that up nicely, and remove comparisons with hardcoded strings

damonkelley replied:
It also contains those "settings" to a single location. If, for whatever reason, we need to change them, we won't have to dig through the code to make it happen.

Commit:
Added class constants to models.py to remove comparisons with hardcoded strings and make changes to the choices simpler.

Commit:
Simplified the is_success() method of models.py.

Commit:
Fixed a remaining colon in the is_success() method.

damonkelley commented:
:+1: :100:
